### ISSUE TITLE
`updated` record being marked as `created`

### ISSUE THREAD

**Issue by MikeGlennFWC** [Nozbe/WatermelonDB#1141]
Hello,

I have a local device record that is `created`, then moments later is `updated` again by the device.

The `created` record hits the server and syncs fine.

Then the record is updated on the local device, but the `_status` column is marked as `created`

This tries to send the record again on next sync as `created`, but this should be in the `updated` list. The server rejects the sync because it finds a duplicate key in the database as it tries to insert with the same ID but updated values.

I have a screenshot showing the status of `created` but it showing 2 columns updated.

Is this something to do with the way i have set up WatermelonDB, or is this a bug? I'm using version 0.23.0

<img width="702" alt="Screenshot 2021-08-17 at 15 16 50" src="https://user-images.githubusercontent.com/58774408/129742753-395ff3f7-f085-4ba1-bd5d-4ce94c75d584.png">

**Comment by radex** [comment]
no, this is expected behavior on WatermelonDB side and a bug on your backend, however it's likely that it's been obscured before and now visible because of 0.23 changes.

record stays created most likely because it was changed between push beginning and completion. however, your server should treat `created` and `updated` records essentially the same way - the distinction is kind of arbitrary at this point. https://github.com/Nozbe/WatermelonDB/blob/master/docs-master/Advanced/Sync.md#implementing-push-endpoint

**Comment by MikeGlennFWC** [comment]
Ok, thanks. I'll update to follow this.

One more thing, in your backend implementations, do you have _changes and _status fields on your backend database?

**Comment by radex** [comment]
> One more thing, in your backend implementations, do you have _changes and _status fields on your backend database?

**Comment by MikeGlennFWC** [comment]
Sorry, I'm over thinking it, of course you don't.

I'm now upserting created records, which fixes the issue.

last question, is WatermelonDB ok with receiving the same record in created changes on a pull, even if it's already created that record? because that will happen in the above instance if it's upserted.

It used to throw a warning, but safely updated it.

**Comment by radex** [comment]
> last question, is WatermelonDB ok with receiving the same record in created changes on a pull, even if it's already created that record?

yes
